eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java

```diff
@@ -213,95 +210,23 @@ private Collection<Dependency> deps() throws IOException {
+        String lclf = left.getClassifier();
+        if (lclf == null) {
+            lclf = "";
         }
-        if (!this.ignoreTransitive) {
-            dependencies = new DcsNoOneHasTransitive(
-                dependencies,
-                dep -> new DcsTransitive(
-                    new DcsDepgraph(
-                        project,
-                        session,
-                        manager,
-                        targetDir.toPath()
-                            .resolve(ResolveMojo.DIR)
-                            .resolve("dependencies-info"),
-                        dep
-                    ),
-                    dep
-                )
-            );
+        String rclf = right.getClassifier();
+        if (rclf == null) {
+            rclf = "";
         }
-        return dependencies;
-    }
-
-    /**
-     * Find the artifact required by this EO XML.
-     *
-     * @param file EO file
-     * @return List of artifact needed
-     * @throws IOException If fails
-     */
-    private static Optional<Dependency> artifact(final Path file) throws IOException {
-        final Collection<String> coords = new XMLDocument(file).xpath(
-            "//meta[head='rt' and part[1]='jvm']/part[2]/text()"
-        );
-        final Optional<Dependency> dep;
-        if (coords.isEmpty()) {
-            dep = Optional.empty();
-        } else if (coords.size() == 1) {
-            final String[] parts = coords.iterator().next().split(":");
-            final Dependency dependency = new Dependency();
-            dependency.setGroupId(parts[0]);
-            dependency.setArtifactId(parts[1]);
-            if (parts.length == 3) {
-                dependency.setVersion(parts[2]);
-                dependency.setClassifier("");
-            } else {
-                dependency.setClassifier(parts[2]);
-                dependency.setVersion(parts[3]);
-            }
-            dependency.setScope("transpile");
-            dep = Optional.of(dependency);
-        } else {
-            throw new IllegalStateException(
-                String.format(
-                    "Too many (%d) dependencies at %s",
-                    coords.size(), new Rel(file)
-                )
-            );
-        }
-        return dep;
-    }
-
-    /**
-     * Dep to coords.
-     *
-     * @param dep The dependency
-     * @return Coords
-     */
-    private static String coords(final Dependency dep) {
-        String ret = "";
-        if (dep.getClassifier().isEmpty()) {
-            ret = String.format(
-                "%s:%s:%s",
-                dep.getGroupId(), dep.getArtifactId(), dep.getVersion()
-            );
-        } else {
-            ret = String.format(
-                "%s:%s:%s:%s",
-                dep.getGroupId(), dep.getArtifactId(), dep.getClassifier(), dep.getVersion()
-            );
-        }
-        return ret;
+        return lclf.equals(rclf)
```

Comment:
@yegor256 
```java
return Objects.equals(left.getClassifier(), right.getClassifier()) && Objects.equals(left.getArtifactId(), right.getArtifactId()) && Objects.equals(left.getGroupId(), right.getGroupId())
```

Reply:
@volodya-lombrozo nope, the tests fail with this

Reply:
@yegor256 But the method is correct:
```java
public class DepTest {
    @Test
    void test() {
        Assertions.assertTrue(
            eqTo(
                dep(null, "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertTrue(
            eqTo(
                dep(null, null, "group"),
                dep(null, null, "group")
            )
        );
        Assertions.assertTrue(
            eqTo(
                dep(null, null, null),
                dep(null, null, null)
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, "artifact", "group"),
                dep("classifier", "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep("classifier", "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, null, "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, null, "group"),
                dep(null, null, null)
            )
        );
    }

    private Dependency dep(String classifier, String artifact, String group) {
        final Dependency dep = new Dependency();
        dep.setClassifier(classifier);
        dep.setArtifactId(artifact);
        dep.setGroupId(group);
        return dep;
    }

    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(left.getClassifier(), right.getClassifier())
            && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }
}
```

Reply:
@volodya-lombrozo try this:

```
Assertions.assertTrue(
            eqTo(
                dep("", "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
```

Reply:
@yegor256 
```java
    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(
            Optional.ofNullable(left.getClassifier()).orElse(""),
            Optional.ofNullable(right.getClassifier()).orElse("")
        ) && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }

```

Reply:
@yegor256 or
```java
    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(
            Objects.toString(left.getClassifier(), ""),
            Objects.toString(right.getClassifier(), "")
        ) && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }
```

Reply:
@volodya-lombrozo do you think this code is more readable than the code we have now in the branch?

Reply:
@yegor256 Yes, I think it is.  Less variables. Less lines

Reply:
@volodya-lombrozo line count is not always the indicator of code readability. This boolean condition is complex as it is. Making it even more complex is unnecessary, I believe.

Reply:
@yegor256 what about redundant variables then?

Reply:
@volodya-lombrozo I fixed the code, it looks more or less OK in the second option you suggested